tools/rosbag_storage/CMakeLists.txt

```diff
@@ -7,7 +7,7 @@ if(NOT WIN32)
 endif()
 
 find_package(console_bridge REQUIRED)
-find_package(catkin REQUIRED COMPONENTS cpp_common roscpp_serialization roscpp_traits rostime roslz4)
+find_package(catkin REQUIRED COMPONENTS cpp_common pluginlib roscpp_serialization roscpp_traits rostime roslz4)
```

Comment:
The dependency on `pluginlib` needs to be declared in the package manifest in order to be installed on Jenkins (and become a dependency of the Debian package). Therefore CI currently fails.

In general a dependency on `pluginlib` will be a bit problematic since currently `pluginlib` is only part of the higher level group "ROS base". When required here it will need to be moved to "ROS core" (see REP [142](http://www.ros.org/reps/rep-0142.html#ros-core)).

Reply:
I'm in favour of that change, as I'd like to use plugins (or at least plugin discovery) in tools like rostopic.

Reply:
That would be "challenging" since `rostopic` is written in Python and `pluginlib` in C++ :wink:

Reply:
Yeah, I think I was thinking that the python plugin discovery stuff that rqt uses was part of `pluginlib` itself, but it's totally not; it's all just [right there in the rqt repo](https://github.com/ros-visualization/rqt/blob/kinetic-devel/rqt_gui/src/rqt_gui/ros_plugin_provider.py). Well never mind that then. :)

In any case, pluginlib is itself a lightweight dependency, isn't it? Is the pocoo stuff it depends on heavy?

Reply:
It's not heavy: mostly `pluginlib` and `class_loader`.

The "problem" is that `pluginlib` depends on `rosconsole` which is in the `ros_comm` repo.
 So if `rosbag_storage` would depend on `pluginlib` that would result in a repository-level circular dependency. To avoid that some parts would likely need to be split out into a separate repo. (In general splitting this repo into multiple smaller ones is beneficial - but also *a lot* of effort...)

Reply:
Declared build/run dependency to `pluginlib` in `package.xml` for now.

Reply:
The repo-level circular dependency doesn't impact our builds, so we'd be happy to proceed with this path if the proposed change is otherwise acceptable, and plan to assist with splitting out {{rosconsole}} in time for the ROS M release of this repository.

Alternatively, I could look at breaking `pluginlib`'s dependency on rosconsole, potentially by just changing its logging over to use the non-ROS `console_bridge` API instead. Is that acceptable?

If not, we can retain the idea of encryption being a pluggable capability while not actually allowing external encryption modules to be plugged in (eg, `setEncryptorPlugin` would take a pointer to an instance rather a plugin name and params).

Let us know how you'd like to proceed here.

Reply:
> if the proposed change is otherwise acceptable

I will add some higher level comment to the PR. But the general feature sounds like a good idea to me.

> ... splitting out {{rosconsole}} in time for the ROS M release of this repository.

I don't think we want the repository level circular dependency. So splitting out `rosconsole` before accepting this feature would be a reasonable path forward.

> Alternatively, I could look at breaking `pluginlib`'s dependency on rosconsole, potentially by just changing its logging over to use the non-ROS `console_bridge` API instead. Is that acceptable?

It might be but you might want to check with the `pluginlib` maintainer on this. There might be problems which I don't foresee atm.